test-sbat.c

```diff
@@ -0,0 +1,186 @@
+	if (!alloc_comp_name_size)
+		return NULL;
+	CHAR8 *alloc_comp_name = AllocatePool((strlen(comp_name)+1)*sizeof(*alloc_comp_name));
+	if (!alloc_comp_name)
+		return NULL;
+	CopyMem(alloc_comp_gen, comp_gen, strlen(comp_gen) + 1);
+	CopyMem(alloc_comp_name_size, comp_name_size, strlen(comp_name_size)+1);
+	CopyMem(alloc_comp_name, comp_name, strlen(comp_name)+1);
+	new_entry->component_generation = alloc_comp_gen;
+	new_entry->component_name_size = alloc_comp_name_size;
+	new_entry->component_name = alloc_comp_name;
+	return new_entry;
+}
+
+// TEST 1: .sbat entry in EFI variable
+// Expected Result: Pass
+int
+test_verify_sbat_valid_sbat_pass(void)
+{
+	char sbat_base[] = "test1,1,SBAT test1,acme,1,testURL\n" \
+			   "test2,2,SBAT test2,acme2,2,testURL2\n";
+	size_t sbat_size = sizeof(sbat_base);
+	char *buffer = NULL;
+	struct sbat sbat = { 0 };
+	assert(parse_sbat(sbat_base, sbat_size, buffer, &sbat) == EFI_SUCCESS);
+
+	list_t sbat_entries;
+	INIT_LIST_HEAD(&sbat_entries);
+	struct sbat_var *test_entry;
+	char comp_name[] = "test1";
+	char comp_gen[] = "1";
+	char comp_name_size[5];
+	sprintf(comp_name_size, "%d", strlen(comp_name)+1);
+	test_entry = create_mock_sbat_var_entry(comp_gen, comp_name_size, comp_name);
+	if (!test_entry)
+		return -1;
+	list_add(&test_entry->list, &sbat_entries);
+
+	// Note: verify_sbat also frees the underlying sbat_entries.
+	assert(verify_sbat(&sbat, &sbat_entries) == EFI_SUCCESS);
+	return 0;
+}
+
+// Test: .sbat component generation higher than SBAT EFI generation
+// Expected result: EFI_SECURITY_VIOLATION
+int
+test_verify_sbat_valid_sbat_reject(void)
+{
+	char sbat_base[] = "test1,1,SBAT test1,acme,1,testURL\n" \
+			   "test2,2,SBAT test2,acme2,2,testURL2\n";
+	size_t sbat_size = sizeof(sbat_base);
+	char *buffer = NULL;
+	struct sbat sbat = { 0 };
+	assert(parse_sbat(sbat_base, sbat_size, buffer, &sbat) == EFI_SUCCESS);
+
+	list_t sbat_entries;
+	INIT_LIST_HEAD(&sbat_entries);
+	struct sbat_var *test_entry;
+	char comp_name[] = "test2";
+	char comp_gen[] = "3";
+	char comp_name_size[5];
+	sprintf(comp_name_size, "%d", strlen(comp_name)+1);
+	test_entry = create_mock_sbat_var_entry(comp_gen, comp_name_size, comp_name);
+	if (!test_entry)
+		return -1;
+	list_add(&test_entry->list, &sbat_entries);
+
+	// Note: verify_sbat also frees the underlying sbat_entries.
+	assert(verify_sbat(&sbat, &sbat_entries) == EFI_SECURITY_VIOLATION);
+	return 0;
+}
+
+void
+test_verify_sbat_invalid_sbat(void)
+{
+	char sbat_base[] = "test1,1,SBAT test1,acme,1,testURL\n" \
+			   "test2,2,SBAT test2,acme2,2,testURL2\n";
+	size_t sbat_size = sizeof(sbat_base);
+	char * buffer = NULL;
+	struct sbat sbat = { 0 };
+	assert(parse_sbat(sbat_base, sbat_size, buffer, &sbat) == EFI_SUCCESS);
+
+	// Test - no SBAT EFI variable entries
+	list_t sbat_entries;
+	INIT_LIST_HEAD(&sbat_entries);
+	assert(verify_sbat(&sbat, &sbat_entries) == EFI_NOT_FOUND);
```

Comment:
should `verify_sbat` return `EFI_NOT_FOUND` in this case? Currently it returns `EFI_SUCCESS`

Reply:
Indeed, or maybe `EFI_INVALID_PARAMETER` ? @aburmash ^

Reply:
Implemented EFI_INVALID_PARAMETER for empty sbat_entries list, going to push a rebased commit soon

Reply:
Sounds good. Will update the test to check for `EFI_INVALID_PARAMETER`. Thanks!